### ISSUE TITLE
Invalid characters crawl into XML report making it invalid

### ISSUE THREAD

**Issue by mlopatkin** [Grigory-Rylov/android-instrumental-test-runner-core#47]
Stacktraces from Kotlin Coroutines sometimes have '\b' (ASCII backspace, 0x08) character in them, e.g. a line like `at \b\b\b(Coroutine boundary.\b(\b)`. Thanks to KXML's relaxed validation when serializing, this ends up in XML data as `at &#8;&#8;&#8;(Coroutine boundary.&#8;(&#8;)`. However this isn't valid XML 1.0, see https://www.w3.org/TR/xml/#char32 (it is valid 1.1 but file header says 1.0) and breaks lots of existing parsers.

As there is no way of preventing such characters from appearing in the stacktraces, some kind of escaping should be done prior to writing them to XML, probably like it is already done for `\0` in XmlTestRunListener.sanitize.
